### PR TITLE
Limit PDF generation max thread to 2

### PR DESCRIPTION
**Pull request by MichaelTran262**
- Also added WAITING state for PDF generation.

Note: Kramerius API allows only two PDF generation at the same time for a single user.

### CODE REVIEW

**Comment by MichaelTran262** [comment]
Nevím jestli `AsyncConfiguration.java` ovlivňuje i jiné procesy mimo `AsyncPDFGenService.java`, který jako jediný má `@Async` anotaci.

File: backend/src/main/java/cz/mzk/fofola/model/pdf/PDFGenState.java [hunk 1]
```diff
@@ -8,6 +8,8 @@
 public enum PDFGenState {
     
     ACTIVE("active"),
+
```
Comment (kazooo) [inline backend/src/main/java/cz/mzk/fofola/model/pdf/PDFGenState.java]: zbytecny new line

File: backend/src/main/java/cz/mzk/fofola/configuration/AsyncConfiguration.java [hunk 2]
```diff
@@ -1,9 +1,25 @@
...
+    public Executor taskExecutor() {
+        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
+        executor.setCorePoolSize(2);
+        executor.setMaxPoolSize(2);
+        executor.setQueueCapacity(500);
+        executor.setThreadNamePrefix("PdfService-");
+        executor.initialize();
+        return executor;
+    }
+
```
Comment (kazooo) [inline backend/src/main/java/cz/mzk/fofola/configuration/AsyncConfiguration.java]: zbytecny new line

File: backend/src/main/java/cz/mzk/fofola/rest/PDFController.java [hunk 3]
```diff
@@ -62,6 +62,7 @@ public ResponseEntity<Resource> getOutputPDFFile(@PathVariable String uuid) thro
     @ResponseStatus(HttpStatus.OK)
     public void startPDFGenerating(@PathVariable String uuid) {
         log.info("Start asynchronous PDF generating for " + uuid);
-        pdfGenService.start(uuid, null);
+        AsyncPDFGenLog genLog = pdfGenService.prepare(uuid);
+        pdfGenService.start(uuid, null, genLog);
```
Comment (kazooo) [inline backend/src/main/java/cz/mzk/fofola/rest/PDFController.java:67]: prosim nechat strucne komentare, ze davas nejdriv do stavu `WAITING` protoze dal muze cekat na dalsi thread nez prejde do `ACTIVE`

File: frontend/src/modules/pdf/PdfFileList.js [hunk 4]
```diff
@@ -19,7 +19,7 @@ export const PdfFileList = () => {
     const dispatch = useDispatch();
     const outputFiles = useSelector(state => getOutputFiles(state));
 
-    const readyToDownload = state => state !== 'EXCEPTION' && state !== 'ACTIVE';
+    const readyToDownload = state => state !== 'EXCEPTION' && state !== 'ACTIVE' && state !== 'WAITING';
```
Comment (kazooo) [inline frontend/src/modules/pdf/PdfFileList.js:22]: pecka :+1:

File: backend/src/main/java/cz/mzk/fofola/configuration/AsyncConfiguration.java [hunk 5]
```diff
@@ -1,9 +1,25 @@
...
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+
+import java.util.concurrent.Executor;
 
 @Configuration
 @EnableAsync
 public class AsyncConfiguration {
+
+    @Bean
+    public Executor taskExecutor() {
```
Comment (kazooo) [inline backend/src/main/java/cz/mzk/fofola/configuration/AsyncConfiguration.java:20]: Tim vytvaris globalni taskExecutor pro celou appku, a to na 2 vlakna, coz urcite pro celou appku neni vhodny. Potrebujes tenhle taskExecutor jen pro generovani PDF, dej beanu nazev a ten specifikuj v @Async PDF generatoru, vic https://www.baeldung.com/spring-async#1-override-the-executor-at-the-method-level

**Review by kazooo — CHANGES_REQUESTED** [review]
Also, vsude parametry/promenne pridej modifier `final` at jsou to konstanty (good practice).

**Pull request closed without merging**